src/components/UserDetails.jsx

```diff
@@ -1,9 +1,64 @@
+import { useState, useRef, useEffect } from "react";
 import { capitalize } from "../utils";
+import { Form } from "react-bootstrap";
 
 function UserDetails({ user }) {
+  const inputRef = useRef(null);
+  const [username, setUsername] = useState("");
+  const [isText, setIsText] = useState(true);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  useEffect(() => {
+    if (inputRef.current && !isText) {
+      inputRef.current.focus();
+      setIsInputFocused(true);
+    }
+
+    const handleClickOutside = (event) => {
+      if (
+        inputRef.current &&
+        !inputRef.current.contains(event.target) &&
+        isInputFocused
+      ) {
+        setIsText(true);
+        setIsInputFocused(false);
+        setUsername("");
+      }
+    };
+
+    document.addEventListener("click", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("click", handleClickOutside);
```

Comment:
out of my ignorance, why remove the listener when the component did unmount?

Reply:
Because, in the best case scenario, is a waste or resources; the worst would be conflicting with any other hypothetical handler. You should clean the camp zone, before leaving ;)